core/src/Dscp/Core/Fees.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE StrictData #-}
+
+-- | Trnsaction fees processing and calculation.
+module Dscp.Core.Fees
+       ( Fees (..)
+       , FeeCoefficients (..)
+       , calculateFee
+       , noFees
+       , precomputeTxSize
+       , precomputePublicationSize
+       ) where
+
+import Codec.Serialise (Serialise (..), serialise)
+import qualified Data.ByteString.Lazy as BSL
+
+import Dscp.Core.Foundation
+import Dscp.Crypto
+
+-- | Amount of transaction fees.
+newtype Fees = Fees { unFees :: Coin }
+    deriving (Eq, Ord, Show, Generic, Hashable, Bounded)
+
+instance Serialise Fees
+
+-- | Setup for fee calculation.
+data FeeCoefficients = FeeCoefficients
+    { fcMultiplier :: Float   -- ^ How much tokens taken per byte of tx?
+    , fcMinimal    :: Coin    -- ^ How much is the base fee?
+    } deriving (Eq, Show)
```

Comment:
Probably we won't return to fees soon again, so I think there should a support for multiple fee policies for forward compatibility. Something like 
```
data FeePolicy
    = LinearFeePolicy FeeCoefficients
    | UknownFeePolicy
```
so that it `LinearFeePolicy coefs` serialises to `"0" <> encode coefs`, and carry this `FeePolicy` around instead of `FeeCoefficients`.